Kickstarter-iOS/Views/Controllers/CreatePasswordViewController.swift

```diff
@@ -1,27 +1,48 @@
 import Library
 import Prelude
+import Result
+import ReactiveSwift
 import UIKit
 
-private enum CreatePasswordRow: CaseIterable {
+enum CreatePasswordRow: CaseIterable {
   case newPassword
   case confirmNewPassword
 }
 
 extension CreatePasswordRow {
+  var placeholder: String {
+    return Strings.signup_input_fields_password()
+  }
+
+  var returnKeyType: UIReturnKeyType {
+    switch self {
+    case .newPassword: return .next
+    case .confirmNewPassword: return .done
+    }
+  }
+
   var title: String {
     switch self {
     case .newPassword: return Strings.New_password()
     case .confirmNewPassword: return Strings.Confirm_password()
     }
   }
+}
 
-  var placeholder: String {
-    return Strings.signup_input_fields_password()
-  }
+@objc protocol CreatePasswordViewControllerType {
+  @objc optional func newPasswordTextFieldChanged(_ sender: UITextField)
+  @objc optional func newPasswordTextFieldDidReturn(_ sender: UITextField)
+  @objc optional func newPasswordConfirmationTextFieldChanged(_ sender: UITextField)
+  @objc optional func newPasswordConfirmationTextFieldDidReturn(_ sender: UITextField)
 }
 
 final class CreatePasswordViewController: UITableViewController {
-  // MARK: - Accessors
+  // MARK: - Properties
+
+  private let viewModel: CreatePasswordViewModelType = CreatePasswordViewModel()
+  private weak var newPasswordTextField: UITextField?
```

Comment:
These `weak var`iables hold references to the UI (just like we previously used to when using XIBs)